packages/nimble-components/src/table/index.ts

```diff
@@ -282,25 +282,9 @@ export class Table<
     }
 
     /** @internal */
-    public async onRowClick(rowIndex: number): Promise<void> {
-        if (this.selectionMode === TableRowSelectionMode.none) {
-            return;
-        }
-
-        const row = this.table.getRowModel().rows[rowIndex];
-        if (!row) {
-            return;
-        }
-
-        const currentSelection = await this.getSelectedRecordIds();
-        if (currentSelection.length === 1 && currentSelection[0] === row.id) {
-            // The clicked row is already the only selected row. Do nothing.
-            return;
-        }
-
-        this.table.toggleAllRowsSelected(false);
-        row.toggleSelected(true);
-        await this.emitSelectionChangeEvent();
+    public onRowClick(rowIndex: number): boolean {
+        void this.updateSelectionOnRowClickAsync(rowIndex);
+        return true;
```

Comment:
If an event-bound function doesn't return `true` then [`preventDefault()` is called on the event](https://www.fast.design/docs/fast-element/declaring-templates#events) (see the callout marked "IMPORTANT"). This was the case even when the `Promise` resolved to `true`. It's probably doing a dumb comparison of the `Promise` to `true`, rather than waiting for the `Promise` to resolve. The row click handler should not be preventing default, so I changed it to call the async function and return `true`.